src/containers/Notifications.js

```diff
@@ -0,0 +1,151 @@
+import React from 'react'
+import { connect } from 'react-redux'
+import { prefetch } from 'react-fetcher'
+import { fetchActivity, markActivityRead, markAllActivitiesRead, navigate, thank } from '../actions'
+import { values, sortBy, isEmpty, contains } from 'lodash'
+import cx from 'classnames'
+import ScrollListener from '../components/ScrollListener'
+import A from '../components/A'
+import truncate from 'html-truncate'
+import { present, humanDate } from '../util/text'
+const { array, bool, func, number, object } = React.PropTypes
+
+@prefetch(({ dispatch }) => dispatch(fetchActivity(20, 0)))
+@connect(state => ({
+  activities: sortBy(values(state.activities), ['created_at']).reverse(),
+  total: Number(state.totalActivities)
+}))
+export default class Notifications extends React.Component {
+
+  static propTypes = {
+    activities: array,
+    pending: bool,
+    dispatch: func,
+    total: number,
+    currentUser: object
+  }
+
+  loadMore = () => {
+    let { total, activities, dispatch, pending } = this.props
+    let offset = activities.length
+    if (!pending && offset < total) {
+      dispatch(fetchActivity(20, offset))
+    }
+  }
+
+  markAllRead = () => {
+    let { dispatch } = this.props
+    dispatch(markAllActivitiesRead())
+  }
+
+  visit = activity => {
+    let { dispatch } = this.props
+    let path = `/p/${activity.post.id}`
+    if (activity.unread) {
+      dispatch(markActivityRead(activity.id))
+      .then(action => {
```

Comment:
in this situation you might as well not even wait for `.then`, because the second action doesn't depend on the first. might be slightly more responsive?